mlflow/sklearn/__init__.py

```diff
@@ -884,12 +944,59 @@ def patched_fit(original, self, *args, **kwargs):
                           for autologging (e.g., specify "fit" in order to indicate that
                           `sklearn.linear_model.LogisticRegression.fit()` is being patched)
         """
+        global _autolog_training_status
         with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
             if t.should_log():
-                return fit_mlflow(original, self, *args, **kwargs)
+                _autolog_training_status = _AutologTrainingStatus(
+                    last_mlflow_run_id=mlflow.active_run().info.run_id,
+                    model_id=id(self)
+                )
+                print(f'DBG: run patched_fit,'
+                      f'run_id={_autolog_training_status.last_mlflow_run_id},'
+                      f'model_id={_autolog_training_status.model_id}.')
+                try:
+                    _autolog_training_status.in_fit_call_scope = True
+                    result = fit_mlflow(original, self, *args, **kwargs)
+                finally:
+                    _autolog_training_status.in_fit_call_scope = False
+                return result
             else:
                 return original(self, *args, **kwargs)
 
+    def patched_predict(original, self, *args, **kwargs):
+        global _autolog_training_status
+        with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
+            if t.should_log():
+                result = original(self, *args, **kwargs)
+                _autolog_training_status.register_predition_result(self, result)
```

Comment:
In `patched_predict`, we need register the prediction result instance into the `_AutologTrainingStatus`.